plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/SQLServerUtils.java

```diff
@@ -342,6 +341,30 @@ public static String changeCreateToAlterDDL(
         return ddl;
     }
 
+
+    @NotNull
+    public static String getObjectDefinitionFunction(
+        @NotNull String systemSchema,
+        @NotNull DBPDataSource dataSource,
+        @NotNull String objectFQN,
+        @NotNull DBRProgressMonitor monitor
+    ) {
+        String quotedObjectFQN = SQLUtils.quoteString(dataSource, objectFQN);
+        return isDriverBabelfish(dataSource.getContainer().getDriver())
+            ? "SELECT definition FROM sys.sql_modules WHERE object_id = (OBJECT_ID(" + quotedObjectFQN + "))"
+            : isSupportsObjectDefinitionFunction(dataSource, monitor)
+                ? "SELECT OBJECT_DEFINITION(OBJECT_ID(" + quotedObjectFQN + "))"
+                : systemSchema + ".sp_helptext '" + objectFQN + "'";
+    }
+
+    public static boolean isSupportsObjectDefinitionFunction(
+        @NotNull DBPDataSource dataSource,
+        @NotNull DBRProgressMonitor monitor
+    ) {
+        return dataSource instanceof SQLServerDataSource sqlServerDataSource &&
+            (sqlServerDataSource.isDataWarehouseServer(monitor) || sqlServerDataSource.isSynapseDatabase());
```

Comment:
Just my own nitpick, you are not supposed to listen to me. I'd prefer something like this instead:

```suggestion
        // v this is a precondition, an early exit is nice here
        if (!(dataSource instanceof SQLServerDataSource sqlServerDataSource)) {
            return false;
        }
        // v this is an actual check
        return sqlServerDataSource.isDataWarehouseServer(monitor) || sqlServerDataSource.isSynapseDatabase();
```

Comments are just for clarity, omit them.